src/cmd/build.js

```diff
@@ -95,11 +95,23 @@ export function safeFileName(name: string): string {
 export class FileFilter {
   filesToIgnore: Array<string>;
 
-  constructor({filesToIgnore}: Object = {}) {
+  constructor(artifactsDir,{filesToIgnore}: Object = {}) {
+    var eliminateArtifactDir =  new String();
+    var buf = artifactsDir;
+    if(typeof buf !== 'undefined' && buf.indexOf('web-ext-artifacts') != -1){
+        eliminateArtifactDir = buf.slice(buf.indexOf('web-ext-artifacts'));
+    }
+    else if(typeof buf !== 'undefined' && buf.slice(-1) === '/'){
+	eliminateArtifactDir = path.join(buf.slice(0,-1));
+    }
+    else if(typeof buf !== 'undefined'){
+	eliminateArtifactDir =  path.join(buf);
+    }
     this.filesToIgnore = filesToIgnore || [
       '**/*.xpi',
       '**/*.zip',
       '**/.*', // any hidden file
+      '**/*'+eliminateArtifactDir,
```

Comment:
Wont Having two For loops, increase time complexity.  Lets take some path which will get failed in **if check** first For loop, and then it comes to second for loop for checking. When it fails in second also i feel we will increase the run time.

I had a thought to remove the below lines from constructor and make it as a separate method so we can increase readability.

> var eliminateArtifactDir =  new String();
>     var buf = artifactsDir;
>     if(typeof buf !== 'undefined' && buf.indexOf('web-ext-artifacts') != -1){
>         eliminateArtifactDir = buf.slice(buf.indexOf('web-ext-artifacts'));
>     }
>     else if(typeof buf !== 'undefined' && buf.slice(-1) === '/'){
>         eliminateArtifactDir = path.join(buf.slice(0,-1));
>     }
>     else if(typeof buf !== 'undefined'){
>         eliminateArtifactDir =  path.join(buf);
>     }
>